### PR TITLE
Docs rework help

### PR DESCRIPTION
**Pull request by TC72**
Changes for the developer documentation, mainly helping out with grammar but also trying to simplify the technical side where possible.

### CODE REVIEW

File: developers.md [hunk 1]
```diff
@@ -151,41 +154,45 @@ pair in our table that we pass into `pickers.new`, for example after `sorter`.
...
-pass in a new function. In this new function we first close the picker with
-`actions.close` and then get the `selection` with `action_state`. Its important
+the actions defined in the function should be attached. In this case it would
+remove the default actions to move the selected item in the picker,
+`move_selection_{next,previous}`. So in most cases you'll want to return `true`.
+If the function does not return anything then an error is thrown. 
+
+The `attach_mappings` function has two parameters, `prompt_bufnr` is the buffer number
+of the prompt buffer, which we can use to get the pickers object and `map` is a table
+we can use to map actions or functions to arbitrary key sequences.
```
Comment (Conni2461) [inline developers.md]: `map` is a function that we can use to map functions / actions to keys. Its not a table.

**Comment by TC72** [comment]
I've fixed calling `map` a table instead of a function.
Also fixed the sentence "that you already have some knowledge", was missing "you".

Could you also check how I explain what's passed to `display` if you use a function?
I think I understand it correctly but am I right in saying `display` won't be passed in the table to the `display` function as it would be a reference to itself?

```
The `display` key is required and is either a string or a `function(tbl)`,
where `tbl` is the table returned by `entry_maker`. So in this example `tbl` would
give our `display` function access to `value` and `ordinal`. (Not `display` as 
that would be a reference to itself.)
```

**Comment by Conni2461** [comment]
I think thats good. The last sentence is wrong tho. 

> (Not `display` as that would be a reference to itself.)

Because you also have access to display itself. But you shouldn't call that function. Or maybe you can if you want to implement display recursive :rofl:

**Comment by Conni2461** [comment]
So lets remove that sentence. Otherwise its good to go imo.

We can always iterate over it later :laughing: 

Also hacktoberfest is coming and people love to fix typos ;)

**Comment by TC72** [comment]
OK, that's done.
I've asked TJ of he can find the drawing I mentioned.

I'll also look at adding screenshots of what should be seen as you run this code.

**Comment by Conni2461** [comment]
Thanks :)

**Comment by Conni2461** [comment]
But lets do that some other time. Otherwise we never merge it :laughing:

**Comment by TC72** [comment]
Oh yes, we'll do that later.

**Pull request merged**